### ISSUE TITLE
JG: Use a programmatic method (DiagrammR/mermaid) to produce Figure 1 

### ISSUE THREAD

**Issue by hturner** [Abhi-1U/texor-rjarticle#12]
Comment from @ajrgodfrey

> Use a programmatic method (DiagrammR/mermaid) to produce Figure 1 
    - This will make the R Markdown source readable (with a screen reader)
    - Also add alt text (via `fig.alt`) to describe the figure in the HTML version.

**Comment by dicook** [comment]
Done. Hopefully this is ok @ajrgodfrey

**Comment by ajrgodfrey** [comment]
You have done what you can. The imperfections I now observe are entirely the failings of screen readers (JAWS and NVDA) to pick the result as a graphic (JAWS) or use the alt text (both) for the workflow representation figure.

I have checked what NVDA reports in the HTML with the raw Rmd and am comfortable enough with the outcome. I can see a nice fig.alt description has been provided but it isn't automatically being used in the HTML.

The choices are to either:

  1.  Use this chart, with the alt text not being picked up by screen reader failures, or
  2.  Revert to the png, but import it using knitr::include_graphics() so the alt text is fully used.

Option 1 is what the future should look like so is a better standard setter, whereas Option 2 delivers an immediately more pleasurable reading experience. Option 1 is clear enough to understand because this is a simple workflow. If the nodes had much richer connections, the Option 1 result as it stands today would be quite inadequate.

N.B. Figure 2 doesn't have an alt text tag at present. Fixing that is in line with Option 2.

So, I think the authors need to decide whether this paper shows the exemplar handling of the old way to import graphics (Option 2) or show the way the journal should be promoting reproducibility (Option 1).

My guess is that Option 2 is pragmatic, while Option 1 is idealistic.
